internal/models/task.go

```
@@ -0,0 +1,67 @@
+package models
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/thatnerdjosh/example-webservices/internal/config"
+)
+
+type TaskRequest struct {
+	Name string `json:"name"`
+}
+
+var (
+	ErrTaskNotFound = errors.New("task was not found")
+	ErrBadData      = errors.New("malformed data provided")
+)
+
+func (t *TaskRequest) Process(cfg *config.TaskConfig, r *http.Request) error {
+	b, _ := io.ReadAll(r.Body)
```

ignoring errors even in a toy project is a bad idea